tests/integration/models/application.spec.js

```diff
@@ -623,13 +623,13 @@ describe('Application Model', function () {
 
 			describe('balena.models.application.generateApiKey()', function () {
 				applicationRetrievalFields.forEach((prop) =>
-					it(`should be able to generate an API key by ${prop}`, function () {
-						return balena.models.application
-							.generateApiKey(this.application[prop])
-							.then(function (apiKey) {
-								expect(_.isString(apiKey)).to.be.true;
-								return expect(apiKey).to.have.length(32);
-							});
+					it(`should be able to generate an API key by ${prop}`, async function () {
+						const apiKey = await balena.models.application.generateApiKey(
+							this.application[prop],
+						);
+
+						expect(apiKey).to.be.a('string');
+						expect(apiKey).to.have.length(32);
```

Comment:
Can we also get the latest api key that we have for this app and check it's name property?
We can also store the last known api key id and use it in the filter so that we can be more confident that the one that we got back is the one that we just created.

Reply:
This will not have `name` field assigned. Only refactored because of the `await` thingy.

Updated the provisioning call for the name check.